library/signed/src/Comparison.qs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import Std.Arrays.Tail, Std.Arrays.Zipped, Std.Arrays.Most, Std.Arrays.Rest;
+import Std.Diagnostics.Fact;
+import Utils.ApplyCCNOTChain;
+
+/// # Summary
+/// Wrapper for signed integer comparison: `result = xs > ys`.
+///
+/// # Input
+/// ## xs
+/// First $n$-bit number
+/// ## ys
+/// Second $n$-bit number
+/// ## result
+/// Will be flipped if $xs > ys$
+operation CompareGTSI(xs : Qubit[], ys : Qubit[], result : Qubit) : Unit is Adj + Ctl {
```

Comment:
Consider changing this to a more generic version that takes an operation that gets applied if condition is met, just like ApplyIfGreaterLE.
Consider making function names consistent with unstable arithmetic and probably addition more comparison functions. I suspect this will eventually reside next to the 'unsigned integer' arithmetic functions from unstable.
(Again, this may be future work)

Reply:
I filed #1874 to track this -- since this is primarily a port, I'm going to merge this in now and then update the API in a smaller follow-on PR.